libs/blocks/table/table.js

```diff
@@ -0,0 +1,430 @@
+    }
+  }
+}
+
+function handleMouseOut(cols) {
+  cols.forEach((e) => {
+    e.classList.remove('hover');
+    e.classList.remove('no-top-border');
+    e.classList.remove('hover-border-bottom');
+  });
+}
+
+function handleMouseOver(cols, table, colNum, isCollapseTable, lastSectionHead, lastExpandIcon) {
+  handleMouseOut(cols);
+
+  const headingRow = table.querySelector('.row-heading');
+  const colClass = `col-${colNum}`;
+  const isLastRowCollapsed = lastExpandIcon?.getAttribute('aria-expanded') === 'false';
+
+  cols.forEach((e) => {
+    if (e.classList.contains('col-highlight') && e.innerText) {
+      const matchingCols = Array.from(e.classList).filter(
+        (className) => className.startsWith(colClass),
+      );
+      matchingCols.forEach((className) => {
+        const noTopBorderCol = headingRow.querySelector(`.${className}`);
+        noTopBorderCol.classList.add('no-top-border');
+      });
+    }
+
+    if (isCollapseTable && isLastRowCollapsed) {
+      const lastSectionHeadCol = lastSectionHead.querySelector(`.col-${colNum}`);
+      lastSectionHeadCol.classList.add('hover-border-bottom');
+    }
+
+    e.classList.add('hover');
+  });
+}
+
+function handleHovering(table) {
+  const row1 = table.querySelector('.row-1');
+  const colsInRowNum = row1.childElementCount;
+  const isMerchTable = table.classList.contains('merch');
+  const startValue = isMerchTable ? 1 : 2;
+  const isCollapseTable = table.classList.contains('collapse');
+  const sectionHeads = table.querySelectorAll('.section-head');
+  const lastSectionHead = sectionHeads[sectionHeads.length - 1];
+  const lastExpandIcon = lastSectionHead.querySelector('.icon.expand');
+
+  for (let i = startValue; i <= colsInRowNum; i++) {
+    const cols = table.querySelectorAll(`.col-${i}`);
+    cols.forEach((e) => {
+      e.addEventListener('mouseover', () => handleMouseOver(cols, table, i, isCollapseTable, lastSectionHead, lastExpandIcon));
+      e.addEventListener('mouseout', () => handleMouseOut(cols));
+    });
+  }
+}
+
+function handleScrollEffect(table, gnavHeight) {
+  const highlightRow = table.querySelector('.row-highlight');
+  const headingRow = table.querySelector('.row-heading');
+
+  const intercept = table.querySelector('.intercept') || document.createElement('div');
+  intercept.className = 'intercept';
+  intercept.setAttribute('data-observer-intercept', '');
+  table.append(intercept);
+  headingRow.insertAdjacentElement('beforebegin', intercept);
+
+  const observer = new IntersectionObserver(([entry]) => {
+    headingRow.classList.toggle('active', !entry.isIntersecting);
+    if (highlightRow) highlightRow.style.top = `${gnavHeight}px`;
+    headingRow.style.top = `${gnavHeight + (highlightRow ? highlightRow.offsetHeight : 0)}px`;
+  });
+  observer.observe(intercept);
+}
+
+function applyStylesBasedOnScreenSize(table, originTable) {
+  const isMerch = table.classList.contains('merch');
+  const desktopSize = 900;
+  const mobileSize = 768;
```

Comment:
nit
```suggestion
  const DESKTOP_SIZE = 900;
  const MOBILE_SIZE = 768;
```
could also be `SIZE_DESKTOP` etc